hoodie-client/src/main/java/com/uber/hoodie/func/CopyOnWriteLazyInsertIterable.java

```diff
@@ -61,20 +59,30 @@ public CopyOnWriteLazyInsertIterable(Iterator<HoodieRecord<T>> sortedRecordItr,
     this.hoodieTable = hoodieTable;
   }
 
+  // Used for caching HoodieRecord along with insertValue. We need this to offload computation work to buffering thread.
+  static class BufferedPayload<T extends HoodieRecord> {
```

Comment:
Can you have a different name for this BufferedPayload class. We need to capture that this class encapsulates the result of converting HoodieRecord to Avro (e.g : HoodieAvroRecordGenResult)

Reply:
Also,  this can be a standalone class

Reply:
Done, changed the name, but like to keep the class here since it's only used for InsertHandlers ?